lib/util/git/index.ts

```diff
@@ -395,6 +398,34 @@ export async function isBranchStale(branchName: string): Promise<boolean> {
   return !branches.all.map(localName).includes(branchName);
 }
 
+export async function isBranchModified(branchName: string): Promise<boolean> {
+  // First check cache
+  if (config.branchIsModified[branchName] !== undefined) {
+    return config.branchIsModified[branchName];
+  }
+  if (!(await branchExists(branchName))) {
+    throw Error(
+      'Cannot check modification for branch that does not exist: ' + branchName
+    );
+  }
+  // Retrieve the author of the most recent commit
+  const lastAuthor = (
+    await git.raw(['log', '-1', '--pretty=format:%ae', `origin/${branchName}`])
+  ).trim();
+  const { gitAuthorEmail } = config;
+  if (lastAuthor === gitAuthorEmail) {
+    // author matches - branch has not been modified
+    config.branchIsModified[branchName] = false;
+    return false;
+  }
+  logger.debug(
+    { branchName, lastAuthor, gitAuthorEmail },
+    'Last commit author does not match git author email - branch has been modified'
+  );
+  config.branchIsModified[branchName] = true;
```

Comment:
Do we need to reset the cache after force rebase a modified PR?

Reply:
Value should be set to `false` after a commit.